src/Chainweb/Pact/RestAPI/Server.hs

```diff
@@ -425,12 +428,16 @@ internalPoll cutR cid chain cut requestKeys0 = do
 toPactTx :: Transaction -> Maybe (Command Text)
 toPactTx (Transaction b) = decodeStrict' b
 
-validateCommand :: Command Text -> Either String ChainwebTransaction
-validateCommand cmdText = case verifyCommand cmdBS of
+validateCommand :: ChainwebVersion -> Command Text -> Either String ChainwebTransaction
+validateCommand v cmdText = case verifyCommand cmdBS of
     ProcSucc cmd
-        | length (_cmdSigs cmd) <= 100 -> Right (mkPayloadWithText <$> cmd)
-        | otherwise -> Left "More than 100 signatures given."
+        | length (_cmdSigs cmd) > 100 -> Left "More than 100 signatures given."
+        | payloadVer cmd /= Just v -> Left "Incompatible ChainwebVersion given."
```

Comment:
FYI -- I put the `ChainId` check into `Chainweb.Chainweb`. They should probably agree, i.e. either you move this check there or you move that check here?

Reply:
I will resolve the locations of these checks in a follow-up, and ensure that they're occurring everywhere they're supposed to be.